### PR TITLE
Replace picard SamToFastq with samtools fastq

### PR DESCRIPTION
**Pull request by FrickTobias**
See discussion in #153. 

**TL;DR:** We can remove picardtools and also get the compatibility to
- add bam tags to header
- use threading
- get less dependencies
- skip getting /1 and /2 written to the end of the header (causes errors in most software)
- probably have more efficient RAM usage, however this is not tested for samtools fastq

### CODE REVIEW

File: src/blr/Snakefile [hunk 1]
```diff
@@ -168,12 +168,13 @@ rule bam_to_fastq:
...
         bam = "mapped.sorted.tag.mkdup.bcmerge.mol.filt.bam"
-    log: "picard_samtofastq.log"
+    log: "samtools-fastq.log"
     shell:
-        "picard -Xms{config[heap_space]}g SamToFastq"
-        " I={input.bam}"
-        " FASTQ={output.r1_fastq}"
-        " SECOND_END_FASTQ={output.r2_fastq} 2>> {log}"
+        "samtools fastq"
+        " -@ {threads}"
```
Comment (pontushojer) [inline src/blr/Snakefile]: For `threads` to be useful we need to define the maximum number of threads for this rule. Add `threads: 20` after `log: "samtools-fastq.log"`

Comment (FrickTobias) [reply src/blr/Snakefile]: I just remembered, it's fixed now.

**Comment by pontushojer** [comment]
One thing, we are getting this message in the test file.

```
+picard SamToFastq --version
tests/run.sh: line 7: picard: command not found
```
You need to remove the line `picard SamToFastq --version` from `tests/run.sh`.

**Comment by FrickTobias** [comment]
Remembered there are also heap space and other arguments that can be removed. Hold tight.

**Comment by FrickTobias** [comment]
Now everything should be removed, sorry about that!

**Comment by marcelm** [comment]
Wow, very nice. I like how everything gets shorter, in particular the list of dependencies required on Linux!

**Pull request merged**